collectors/mimecast/collector.js

```diff
@@ -0,0 +1,204 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2020, Alert Logic, Inc
+ * @doc
+ *
+ * mimecast class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const calcNextCollectionInterval = require('@alertlogic/paws-collector').calcNextCollectionInterval;
+const packageJson = require('./package.json');
+const utils = require("./utils");
+
+
+let typeIdPaths = [];
+
+let tsPaths = [];
+
+const Siem_Logs = 'SiemLogs';
+
+
+class MimecastCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, packageJson.version);
+    }
+
+    pawsInitCollectionState(event, callback) {
+        const startTs = process.env.paws_collection_start_ts ?
+            process.env.paws_collection_start_ts :
+            moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+
+        const applicationNames = JSON.parse(process.env.collector_streams);
+        const initialStates = applicationNames.map(stream => {
+            if (stream === Siem_Logs) {
+                return {
+                    stream,
+                    nextPage: null,
+                    poll_interval_sec: 1
+                }
+            }
+            else {
+                return {
+                    stream,
+                    since: startTs,
+                    until: endTs,
+                    nextPage: null,
+                    poll_interval_sec: 1
+                }
+            }
+        });
+        return callback(null, initialStates, 1);
+    }
+
+    pawsGetRegisterParameters(event, callback) {
+        const regValues = {
+            mimecastApplicationNames: process.env.collector_streams,
+            mimecastApplicationID: process.env.paws_collector_param_string_1,
```

Comment:
please do not pass secrets as registration params. we don't want to store them in the backend

Reply:
Is it actually a secret?

Reply:
For execute mimecast API we want accessKey(clientID), secretKey(paws secret key), appId (application ID) and appKey (application key).